app/src/main/java/com/concordium/wallet/ui/cis2/TokensAddAdapter.kt

```diff
@@ -49,17 +49,28 @@ class TokensAddAdapter(
         val token = dataSet[position]
         val tokenMetadata = token.tokenMetadata
 
-        if (tokenMetadata?.thumbnail != null && !tokenMetadata.thumbnail.url.isNullOrBlank()) {
-            Glide.with(context)
-                .load(tokenMetadata.thumbnail.url)
-                .placeholder(R.drawable.ic_token_loading_image)
-                .override(iconSize)
-                .fitCenter()
-                .error(R.drawable.ic_token_no_image)
-                .into(holder.binding.tokenIcon)
-        } else {
-            holder.binding.tokenIcon.setImageResource(R.drawable.ic_token_no_image)
-        }
+        tokenMetadata?.let {
+            if (tokenMetadata.thumbnail?.url?.isNotBlank() == true) {
+                Glide.with(context)
+                    .load(tokenMetadata.thumbnail.url)
+                    .placeholder(R.drawable.ic_token_loading_image)
+                    .override(iconSize)
+                    .fitCenter()
+                    .error(R.drawable.ic_token_no_image)
+                    .into(holder.binding.tokenIcon)
+
+            }
+
+            if (tokenMetadata.display?.url?.isNotBlank() == true) {
```

Comment:
This should be an `else if` as we don't want to overwrite the thumbnail if there is a thumbnail.